### PR TITLE
Refactor and Improve Model Manager Implementation

### PR DESCRIPTION
**Pull request by dorbanianas**
This pull request introduces a comprehensive refactor of the `model_manager`, streamlining its functionality and resolving legacy issues. Key changes include:

- Removal of the old `model_manager` structure.
- Adaptation of the project to the new `model_manager`.

### CODE REVIEW

File: src/include/flockmtl/model_manager/providers/handlers/azure.hpp [hunk 1]
```diff
@@ -33,8 +31,8 @@ class AzureModelManager {
     }
 
     nlohmann::json CallEmbedding(const nlohmann::json &json, const std::string &contentType = "application/json") {
-        string url = "https://" + _resource_name + ".openai.azure.com/openai/deployments/" + _deployment_model_name +
-                     "/embeddings?api-version=" + _api_version;
+        std::string url = "https://" + _resource_name + ".openai.azure.com/openai/deployments/" +
```
Comment (queryproc) [inline src/include/flockmtl/model_manager/providers/handlers/azure.hpp:34]: Should API Version be fixed as a const?

File: src/model_manager/model.cpp [hunk 2]
```diff
@@ -0,0 +1,100 @@
+#include "flockmtl/model_manager/model.hpp"
+
+namespace flockmtl {
+
+Model::Model(const nlohmann::json &model_json) {
+    // TODO: remove this when refactoring the aggregate functions
```
Comment (queryproc) [inline src/model_manager/model.cpp:6]: Let's never have TODOs in code and add them as issues in Github

**Review by queryproc — APPROVED** [review]

File: src/include/flockmtl/model_manager/providers/handlers/azure.hpp (hunk 1 above)
Comment (dorbanianas) [reply src/include/flockmtl/model_manager/providers/handlers/azure.hpp:34]: I'm not sure about this is Sunny's implementation, and I still need to go through the providers' implementation and test them

File: src/model_manager/model.cpp (hunk 2 above)
Comment (dorbanianas) [reply src/model_manager/model.cpp:6]: sure I'll do that

File: src/include/flockmtl/model_manager/providers/provider.hpp [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    ModelDetails model_details_;
+
+    explicit IProvider(const ModelDetails &model_details) : model_details_(model_details) {};
+    virtual ~IProvider() = default;
+
+    virtual nlohmann::json CallComplete(const std::string &prompt, bool json_response) = 0;
+    virtual nlohmann::json CallEmbedding(const std::vector<std::string> &inputs) = 0;
+};
+
+class LengthExceededError : public std::exception {
```
[the remark this answers is not in the document]
Comment (dorbanianas) [reply src/include/flockmtl/model_manager/providers/provider.hpp]: done

**Pull request merged**